backend/server.py

```diff
@@ -787,21 +801,25 @@ def delete_user(user_id):
         tuple: A tuple containing a JSON response and a status code.
     """
     # Check connection to the database
-    try:
-        connect_to_db()
-    except Exception as e:
-        print("Error connecting to the database:", str(e))
-        return (
-            jsonify({"error": "Database connection error"}),
-            status.HTTP_500_INTERNAL_SERVER_ERROR,
-        )
+    try_connect_to_db()
 
     try:
         # Connect to the database
         db = firestore.client()
 
-        # Delete the post from the 'posts' collection
-        db.collection("users").document(user_id).delete()
+        # Get the user reference
+        user_ref = db.collection("users").document(user_id)
+
+        # Get the user data
+        user_data = user_ref.get().to_dict()
+
+        # Delete all posts from the user
+        for post in user_data["posts"]:
+            post_ref = db.document(post.path)

```

Comment:
post.path here needs to be changed to just post as this is currently giving me an error when trying to delete a user from the database.

Reply:
I looked into this further and it seems the issue does not rely here. The issue relies with the create_user function as it is saving the data as strings instead of saving the data as references. Please modify create_user function and verify that the create_user() and update_user() functions save the data as references and not as strings.

Reply:
I've hardcoded the desired values for create_user because when a new user is created all schema should be initialized to its null equivalent